src/main/java/uk/gov/hmcts/reform/slc/services/ReportParser.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.slc.services;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.slc.model.LetterPrintStatus;
+import uk.gov.hmcts.reform.slc.services.steps.getpdf.FileNameHelper;
+import uk.gov.hmcts.reform.slc.services.steps.sftpupload.ParsedReport;
+import uk.gov.hmcts.reform.slc.services.steps.sftpupload.Report;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.StreamSupport.stream;
+
+@Component
+public class ReportParser {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportParser.class);
+
+    public ParsedReport parse(Report report) {
+        try (CSVParser parser = parserFor(report.content)) {
+
+            List<LetterPrintStatus> statuses =
+                stream(parser.spliterator(), false)
+                    .map(row -> toPrintStatus(row))
+                    .filter(Objects::nonNull)
+                    .collect(toList());
+
+            return new ParsedReport(report.path, statuses);
```

Comment:
i think, in theory, since we only use parsed report internally, is it more efficient to use `statuses` as stream so when it gets parsed in the task runner it'll just use same parallel/sequential stream to foreach the elements.

does that make sense? will it just make things more confusing? can be reviewed at the later stage

Reply:
I agree but I copied this class over from the consumer service; are you happy for me to change it in a follow up?

Reply:
oh, definitely not this one. observation on the whole gave a thought